src/rp2_common/hardware_pio/include/hardware/pio.h

```diff
@@ -215,6 +215,10 @@ static_assert(DREQ_PIO0_TX2 == DREQ_PIO0_TX0 + 2, "");
 static_assert(DREQ_PIO0_TX3 == DREQ_PIO0_TX0 + 3, "");
 static_assert(DREQ_PIO0_RX0 == DREQ_PIO0_TX0 + NUM_PIO_STATE_MACHINES, "");
 static_assert(DREQ_PIO1_RX0 == DREQ_PIO1_TX0 + NUM_PIO_STATE_MACHINES, "");
+#if NUM_PIOS > 2
+static_assert(DREQ_PIO2_TX0 == DREQ_PIO1_RX0 + NUM_PIO_STATE_MACHINES, "");
```

Comment:
For "symmetry", perhaps it's also worth adding:
```c
static_assert(DREQ_PIO1_TX0 == DREQ_PIO0_RX0 + NUM_PIO_STATE_MACHINES, "");
```
just above the `static_assert(DREQ_PIO1_RX0 ...` line?